### PR TITLE
Add affine shifts for linear operators, update documentation Changes:

### PR DESCRIPTION
**Pull request by tamiko**
2e56765 (Matthias Maier, 24 minutes ago)
   Update documentation

6b3784d (Matthias Maier, 45 minutes ago)
   Add a notion of affine linear operators

1182efb (Matthias Maier, 54 minutes ago)
   Add operator*=(number) variant to LinearOperator

### CODE REVIEW

File: doc/doxygen/headers/laoperators.h [hunk 1]
```diff
@@ -65,7 +65,42 @@
...
+ * op_a * k;
+ *
+ * // in-place variants
+ * op_a += op_b;
+ * op_a -= op_b;
+ * op_a *= k;
+ *
+ * // operator composition
+ * op_a * op_b;
+ * op_a *= op_b; // If op_b is an endomorphism of the domain space of op_a
```
Comment (bangerth) [inline doc/doxygen/headers/laoperators.h]: why the comment? why would this not be equivalent to
```
  *this = *this  *  op_b
```
?

File: doc/doxygen/headers/laoperators.h [hunk 2]
```diff
@@ -65,7 +65,42 @@
...
+ * Further, by adding (or subtracting) a vector of the range space to
+ * (from) a linear operator, an affine variant is created:
+ * @code
+ * dealii::LinearOperator<> op_a;
+ * dealii::Vector<double> offset;
+ *
+ * op_a + offset;
+ * op_a + offset;
+ * offset - op_a;
+ * offset - op_a;
```
Comment (bangerth) [inline doc/doxygen/headers/laoperators.h]: I'm not sure I'm so enthusiastic about these affine variants. How do they interact with the other operations? For example, what does
```
  (op + vec) * k
```
yield? How about
```
  (op1 + vec) * op2
```
? Does this yield something different than
```
  op1 * (op2 + vec)
```
?

**Comment by bangerth** [comment]
I'm not excited about the idea of affine operators under the name "linear" operators. I find that rather confusing. Why do you need this?

The other parts of the patch are non-controversial.

**Comment by tamiko** [comment]
Nevermind.

**Comment by tamiko** [comment]
Documentation update is #880

**Pull request closed without merging**